ngraph/test/frontend/frontend_manager.cpp

```diff
@@ -161,3 +162,113 @@ TEST(FrontEndManagerTest, testDefaultPlace)
+}
+
+TEST(FrontEndExceptionTest, frontend_general_error_throw_no_info)
+{
+    EXPECT_THROW(FRONT_END_GENERAL_CHECK(false), ngraph::frontend::GeneralFailure);
+}
+
+TEST(FrontEndExceptionTest, frontend_initialization_error_no_throw)
+{
+    EXPECT_NO_THROW(FRONT_END_INITIALIZATION_CHECK(true));
+}
+
+TEST(FrontEndExceptionTest, frontend_initialization_error_no_throw_info)
+{
+    EXPECT_NO_THROW(FRONT_END_INITIALIZATION_CHECK(true, "msg example"));
+}
+
+TEST(FrontEndExceptionTest, frontend_initialization_error_throw_no_info)
+{
+    EXPECT_THROW(FRONT_END_INITIALIZATION_CHECK(false), ngraph::frontend::InitializationFailure);
+}
+
+TEST(FrontEndExceptionTest, frontend_op_conversion_error_no_throw)
+{
+    EXPECT_NO_THROW(FRONT_END_OP_CONVERSION_CHECK(true));
+}
+
+TEST(FrontEndExceptionTest, frontend_op_conversion_error_no_throw_info)
+{
+    EXPECT_NO_THROW(FRONT_END_OP_CONVERSION_CHECK(true, "msg example"));
+}
+
+TEST(FrontEndExceptionTest, frontend_op_conversion_error_throw_no_info)
+{
+    EXPECT_THROW(FRONT_END_OP_CONVERSION_CHECK(false), ngraph::frontend::OpConversionFailure);
+}
+
+TEST(FrontEndExceptionTest, frontend_assert_throw_check_info)
+{
+    std::string msg("msg example");
+    try {
+        FRONT_END_THROW(msg);
+    } catch (const ngraph::frontend::GeneralFailure &ex) {
+        std::string caught_msg(ex.what());
+        EXPECT_NE(caught_msg.find(msg), std::string::npos);
+    } catch (...) {
+        FAIL() << "Not expected exception type.";
+    }
+}
+
+TEST(FrontEndExceptionTest, frontend_not_implemented_throw_check_info)
+{
+    struct TestClass {};
+    try {
+        FRONT_END_NOT_IMPLEMENTED(TestClass);
+    } catch (const ngraph::frontend::NotImplementedFailure &ex) {
+        std::string caught_msg(ex.what());
+        EXPECT_NE(caught_msg.find("TestClass"), std::string::npos);
+    } catch (...) {
+        FAIL() << "Not expected exception type.";
+    }
+}
+
+TEST(FrontEndExceptionTest, frontend_general_error_throw_info)
+{
+    std::string msg("msg example");
+    try {
+        EXPECT_NO_THROW(FRONT_END_GENERAL_CHECK(true, msg));
+    } catch (const ngraph::frontend::GeneralFailure &ex) {
+        std::string caught_msg(ex.what());
+        EXPECT_NE(caught_msg.find(msg), std::string::npos);
+    } catch (...) {
+        FAIL() << "Not expected exception type.";
+    }
+}
+
+TEST(FrontEndExceptionTest, frontend_op_conversion_error_throw_info)
+{
+    std::string msg("msg example");
+    try {
+        EXPECT_NO_THROW(FRONT_END_OP_CONVERSION_CHECK(true, msg));
+    } catch (const ngraph::frontend::OpConversionFailure &ex) {
+        std::string caught_msg(ex.what());
+        EXPECT_NE(caught_msg.find(msg), std::string::npos);
+    } catch (...) {
+        FAIL() << "Not expected exception type.";
+    }
+}
+
+TEST(FrontEndExceptionTest, frontend_initialization_error_throw_info)
+{
+    std::string msg("msg example");
+    try {
+        EXPECT_NO_THROW(FRONT_END_INITIALIZATION_CHECK(true, msg));
```

Comment:
Why do we have `EXPECT_NO_THROW` inside `try...catch` block?

Reply:
fixed